src/mediatek_pkg.c

```diff
@@ -330,8 +330,10 @@ void extract_lzhs_fs(MFILE *mf, const char *dest_file, config_opts_t *config_opt
 		asprintf(&outSeg, "%s/%s.%u", tmpdir, base, i);
 
 		MFILE *seg = mopen(outSeg, O_RDONLY);
-		fwrite(mdata(seg, void), msize(seg), 1, out_file);
-		mclose(seg);
+		if(seg) {
+			fwrite(mdata(seg, void), msize(seg), 1, out_file);
+			mclose(seg);
+		}
```

Comment:
Could you add an error message to let users know that a package wasn't written to disk because of a memory allocation failure?